moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planner_interface.cpp

```diff
@@ -0,0 +1,275 @@
+          result->error_message = reaction_result.error_message;
+          hybrid_planning_goal_handle_->abort(result);
+          RCLCPP_ERROR(LOGGER, "Hybrid Planning Manager failed to react to  '%s'", reaction_result.event.c_str());
+        }
+      });
+}
+
+bool HybridPlannerInterface::sendGlobalPlannerAction()
+{
+  auto global_goal_options = rclcpp_action::Client<moveit_msgs::action::GlobalPlanner>::SendGoalOptions();
+
+  // Add goal response callback
+  global_goal_options.goal_response_callback =
+      [this](std::shared_future<rclcpp_action::ClientGoalHandle<moveit_msgs::action::GlobalPlanner>::SharedPtr> future) {
+        auto const& goal_handle = future.get();
+        auto planning_progress = std::make_shared<moveit_msgs::action::HybridPlanner::Feedback>();
+        auto& feedback = planning_progress->feedback;
+        if (!goal_handle)
+        {
+          feedback = "Global goal was rejected by server";
+        }
+        else
+        {
+          feedback = "Global goal accepted by server";
+        }
+        hybrid_planning_goal_handle_->publish_feedback(planning_progress);
+      };
+  // Add result callback
+  global_goal_options.result_callback =
+      [this](const rclcpp_action::ClientGoalHandle<moveit_msgs::action::GlobalPlanner>::WrappedResult& global_result) {
+        // Reaction result from the latest event
+        ReactionResult reaction_result =
+            ReactionResult(HybridPlanningEvent::UNDEFINED, "", moveit_msgs::msg::MoveItErrorCodes::FAILURE);
+        switch (global_result.code)
+        {
+          case rclcpp_action::ResultCode::SUCCEEDED:
+            reaction_result = planner_logic_->react(HybridPlanningEvent::GLOBAL_PLANNING_ACTION_SUCCESSFUL);
+            break;
+          case rclcpp_action::ResultCode::CANCELED:
+            reaction_result = planner_logic_->react(HybridPlanningEvent::GLOBAL_PLANNING_ACTION_CANCELED);
+            break;
+          case rclcpp_action::ResultCode::ABORTED:
+            reaction_result = planner_logic_->react(HybridPlanningEvent::GLOBAL_PLANNING_ACTION_ABORTED);
+            break;
+          default:
+            break;
+        }
+        // Abort hybrid planning if reaction fails
+        if (reaction_result.error_code.val != moveit_msgs::msg::MoveItErrorCodes::SUCCESS)
+        {
+          auto result = std::make_shared<moveit_msgs::action::HybridPlanner::Result>();
+          result->error_code.val = reaction_result.error_code.val;
+          result->error_message = reaction_result.error_message;
+
+          hybrid_planning_goal_handle_->abort(result);
+          RCLCPP_ERROR(LOGGER, "Hybrid Planning Manager failed to react to  '%s'", reaction_result.event.c_str());
+        }
+      };
+
+  // Forward global trajectory goal from hybrid planning request TODO(sjahr) pass goal as function argument
+  auto global_goal_msg = moveit_msgs::action::GlobalPlanner::Goal();
+  global_goal_msg.motion_sequence =
```

Comment:
This line causes a segfault

Reply:
Where is the pointer `hybrid_planning_goal_handle_` initalized?